client/components/navigation/NavBar.jsx

```diff
@@ -1,29 +1,108 @@
-import React from 'react';
-import 'bootstrap/dist/css/bootstrap.css';
-// import { render } from 'react-dom';
-
-
-const NavBar = () => (
-  <div>
-    <nav className="navbar navbar-light" style={{backgroundColor: '#e3f2fd'}}>
-      <a className="navbar-brand" href="#">
-        PushIt
-      </a>
-      <div className="collapse navbar-collapse" id="navbarText">
-        <ul className="navbar-nav mr-auto">
-          <li className="nav-item active">
-            <a className="nav-link" href="#">Home <span className="sr-only">(current)</span></a>
-          </li>
-          <li className="nav-item">
-            <a className="nav-link" href="#">Features</a>
-          </li>
-          <li className="nav-item">
-            <a className="nav-link" href="#">Pricing</a>
-          </li>
-        </ul>
+import React, { Component } from 'react';
+import PropTypes, { shape } from 'prop-types';
+import toastr from 'toastr';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import './nav.scss';
+import { getCurrentUser } from '../../actions/userActions';
+
+
+export class NavBar extends Component {
+  state = {
```

Comment:
we can do without state here
just grab user from this.props in render function
`const { user: { name }  = {} } = this.props`

BTW, this `{} || this.props.user` will always evaluate to {}